src/codeflare_sdk/cluster/widgets.py

```diff
@@ -89,3 +100,234 @@ def is_notebook() -> bool:
         return True
     else:
         return False
+
+
+def view_clusters(namespace: str = None):
+    """view_clusters function will display existing clusters with their specs, and handle user interactions."""
+    from .cluster import get_current_namespace
+    if not namespace:
+        namespace = get_current_namespace()
+
+    user_output = widgets.Output()
+    raycluster_data_output = widgets.Output()
+    url_output = widgets.Output()
+
+    df = _fetch_cluster_data(namespace)
```

Comment:
nit: I would be against a generic naming scheme for the data frame. This could easily cause confusion. 
Can you update the name to something like `ray_clusters_df`

Reply:
I agree, thanks! - Updated.